src/utils/multisig.ts

```diff
@@ -170,37 +170,96 @@ export function isDevKeyChangeDetailed(oldConfig: any, newConfig: any): boolean
   return isKeyChangeDetailed(oldConfig, newConfig, 'devKeys')
 }
 
+/**
+ * Normalizes an Ethereum address to a consistent format for comparison
+ * 
+ * @param address The Ethereum address to normalize
+ * @returns The normalized Ethereum address
+ */
+export function normalizeEthAddress(address: string): string {
+  if (!address) {
+    return '';
+  }
+
+  let normalized = address.trim();
+  
+  // Add 0x prefix if missing
+  if (!normalized.toLowerCase().startsWith('0x')) {
+    normalized = '0x' + normalized;
+  } else if (normalized.startsWith('0X')) {
+    // Ensure consistent 0x prefix (not 0X)
+    normalized = '0x' + normalized.substring(2);
+  }
+
+  // Convert to lowercase for comparison purposes
+  // This preserves the address format but allows for case-insensitive comparison
+  return normalized.toLowerCase();
+}
+
 /**
  * Removes keys from multiSigPermissions that are not in currentConfig.debug.multisigKeys
+ * This version normalizes addresses for better matching across case differences and formatting.
  *
  * @param multiSigPermissions The multiSigPermissions object containing lists of permitted keys
  * @param currentConfig The current configuration object containing valid multisig keys
  * @returns A new multiSigPermissions object with invalid keys removed
  */
 export function cleanMultiSigPermissions(multiSigPermissions: any, currentConfig: any): any {
-  // If either input is invalid, return the original permissions
-  if (!multiSigPermissions || !currentConfig?.debug?.multisigKeys) {
-    return multiSigPermissions
+  // Check if either param is falsy
+  if (!multiSigPermissions || !currentConfig || !currentConfig.debug || !currentConfig.debug.multisigKeys) {
+    return multiSigPermissions;
```

Comment:
I think prettier is not enabled... it would have removed the `;`

<img width="557" alt="image" src="https://github.com/user-attachments/assets/a3641797-b27e-4316-86d4-6a5a16bec74d" />

Reply:
Yeah prettier checks are not required on this repo =S

Reply:
ran prettier scoped to this file only